cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AuthorizeController.java

```diff
@@ -105,30 +120,49 @@ public ModelAndView handleRequestInternal(final HttpServletRequest request, fina
         return redirectToApproveView(registeredService, context);
     }
 
-    private String buildCallbackUrlForImplicitResponseType(final String state, final Authentication authentication, 
-                                                             final Service service, final String redirectUri) {
-        final AccessToken accessToken = generateAccessToken(service, authentication);
+    private String buildCallbackUrlForImplicitResponseType(final J2EContext context,
+                                                           final Authentication authentication,
+                                                           final Service service,
+                                                           final String redirectUri) {
+
+        final String state = authentication.getAttributes().get(OAuthConstants.STATE).toString();
+        final String nonce = authentication.getAttributes().get(OAuthConstants.NONCE).toString();
+
+        final AccessToken accessToken = generateAccessToken(service, authentication, context);
         logger.debug("Generated Oauth access token: {}", accessToken);
 
         String callbackUrl = redirectUri;
-        callbackUrl += "#access_token=" + accessToken.getId() + "&token_type=bearer&expires_in=" + this.timeout;
-        if (state != null) {
-            callbackUrl += "&state=" + EncodingUtils.urlEncode(state);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Prefer StringBuffer over += for concatenating strings](https://www.codacy.com/app/mmoayyed/cas/file/2522747891/issues/source?bid=3256115&fileBranchId=3273836#l135)